source/fe/fe_system.cc

```diff
@@ -2201,6 +2201,73 @@ FESystem<dim,spacedim>::get_constant_modes () const
 
 
 template <int dim, int spacedim>
+void
+FESystem<dim,spacedim>::
+convert_generalized_support_point_values_to_nodal_values (const std::vector<Vector<double> > &support_point_values,
+                                                          std::vector<double>                &nodal_values) const
+{
+  // we currently only support the case where each base element has exactly
+  // as many generalized support points as there are dofs in that element.
+  //
+  // in the more general case, if there are more generalized support
+  // points than dofs, we don't have the infrastructure at the time of
+  // writing this to tease apart which support point value
+  // belongs to which element
+  AssertDimension (support_point_values.size(), this->dofs_per_cell);
+  AssertDimension (nodal_values.size(), this->dofs_per_cell);
+
+  // loop over the bases and let them do the work on their
+  // share of the input argument
+  std::vector<Vector<double>> base_support_point_values;
+  std::vector<double>         base_nodal_values;
+  unsigned int                current_vector_component = 0;
+  for (unsigned int base=0; base<base_elements.size(); ++base)
+    {
+      const unsigned int element_multiplicity = this->element_multiplicity(base);
+      for (unsigned int m=0;
+           m<element_multiplicity;
+           ++m, current_vector_component += base_element(base).n_components())
+        {
+          // extract the support points of this base element.
+          // assume that support points are numbered in exactly
+          // the same way as dofs -- this is how the
+          // initialize_unit_support_points() function does it,
+          // at least
+          //
+          // to pick things apart, we could really use a base_to_system_index()
+          // function, but that doesn't exist -- just do it by using
+          // the reverse table -- the amount of work done here is not
+          // worth trying to optimizing this
+          base_support_point_values.resize  (base_element(base).dofs_per_cell);
+          for (unsigned int i=0; i<this->dofs_per_cell; ++i)
+            if (this->system_to_base_index(i).first == std::make_pair(base,m))
+              {
```

Comment:
since `get_generalized_support_points` for `FESystem` is not yet implemented, couldn't we simply assume that generalized support points are returned *in blocks*, i.e., base element per base element? In this way it would be very simple to tear them apart (and also to implement the `get_generalized_support_points` in this case).

Without assuming any particular ordering w.r.t. dofs, we could simply define the `generalized_support_points` of the FESystem as the concatenation of all `generalized_support_points` of each base. This saves a lot of time in reordering here, and makes it possible to use more points than dofs right away. I checked and could not find a place where the number of generalized support points are assumed to be the same size of the dofs. 

In fact, FE_RaviarThomas of degree 2 has more dofs than generalized support points... (FE_RaviarThomasNodal, instead, has the same number for both), and no tests seem to be broken by this...